opnode/l2/driver.go

```diff
@@ -2,88 +2,203 @@ package l2
 
 import (
 	"context"
+	"encoding/binary"
 	"fmt"
+	"math/big"
 
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
 	"github.com/ethereum/go-ethereum/trie"
 	"github.com/holiman/uint256"
 )
 
-type InputItem struct {
-	Block    *types.Block
-	Receipts []*types.Receipt
-}
+var (
+	DepositEventABI     = "TransactionDeposited(address,address,uint256,uint256,bool,bytes)"
+	DepositEventABIHash = crypto.Keccak256Hash([]byte(DepositEventABI))
+	DepositContractAddr = common.HexToAddress("0xdeaddeaddeaddeaddeaddeaddeaddeaddead0001")
+	L1InfoFuncSignature = "setL1BlockValues(uint256 _number, uint256 _timestamp, uint256 _basefee, bytes32 _hash)"
+	L1InfoFuncBytes4    = crypto.Keccak256([]byte(L1InfoFuncSignature))[:4]
+	L1InfoPredeployAddr = common.HexToAddress("0x4242424242424242424242424242424242424242")
+)
 
-func (item *InputItem) Epoch() uint64 {
-	return item.Block.NumberU64()
-}
+// UnmarshalLogEvent decodes an EVM log entry emitted by the deposit contract into typed deposit data.
+// parse log data for:
+//     event TransactionDeposited(
+//    	 address indexed from,
+//    	 address indexed to,
+//    	 uint256 value,
+//    	 uint256 gasLimit,
+//    	 bool isCreation,
+//    	 data data
+//     );
+func UnmarshalLogEvent(blockNum uint64, txIndex uint64, ev *types.Log) (*types.DepositTx, error) {
```

Comment:
A general comment, I don't think it was quite necessary to change my code so much for stylistical reasons. It's alright, and definitely no hard feelings, but some people might not find this cool (and I sometimes like it much less, here I think it's relatively minor so whatever).